flycheck.el

```diff
@@ -8103,6 +8103,24 @@ for more information about the custom directories."
                                                  "--print-config" "."))))
     (eq exitcode 0)))
 
+(defun flycheck-eslint--find-working-directory (_checker)
+  "Look for a working directory to run CHECKER in.
+This will either be the directory that contains `.eslintrc' by supports
+configuration files formats, if no such file is found in the directory
+hierarchy, searches `node_modules' directory to detect root of project."
+  (let* ((regex-opt (concat "\\`\\.eslintrc"
+                            "\\(\\.\\(js\\|ya?ml\\|json\\)\\)?\\'")))
+    (or
+     (locate-dominating-file
+      default-directory
+      (lambda (directory)
+        (> (length (directory-files directory nil regex-opt t)) 0)))
+     (and
+      buffer-file-name
+      (or (locate-dominating-file buffer-file-name "node_modules")
+          (file-name-directory buffer-file-name)))
+     default-directory)))
```

Comment:
Sorry, that's not what I has in mind. I was thinking of something like this:

```elisp
(when buffer-file-name
  (or (locate-dominating-file
       buffer-file-name
       (lambda (directory)
         (> (length (directory-files directory nil regex-opt t)) 0)))
      (locate-dominating-file
       buffer-file-name
       "node_modules")))
```

(you don't need default-directory, since that function can return `nil` to indicate that)

Reply:
1) `(locate-dominating-file buffer-file-name (lambda (directory)...` not works because require a directory, should be `(locate-dominating-file (file-name-directory buffer-file-name) (lambda (directory)...`

2) That is the problem, `default-directory` is not being established as working directory, in other parts of the code, the function to set `working-directory` returns a `default-directory` https://github.com/flycheck/flycheck/blob/master/flycheck.el#L6902

Reply:
1. Thanks.
2. I may be missing something, but I don't understand this :/

Reply:
After some test `default-directory` is not necessary